tests/unit/browser/test_adblock.py

```diff
@@ -0,0 +1,434 @@
+# TODO See utils/test_standarddirutils for OSEError and caplog assertion
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+#!/usr/bin/env python3
+
+# Copyright 2014-2015 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutebrowser.browser.adblock."""
+
+import os
+import zipfile
+import shutil
+
+import pytest
+
+from PyQt5.QtCore import pyqtSignal, QUrl, QObject
+
+from qutebrowser.browser import adblock
+from qutebrowser.utils import objreg
+from qutebrowser.commands import cmdexc
+
+
+WHITELISTED_HOSTS = ('qutebrowser.org', 'mediumhost.io')
+
+BLOCKLIST_HOSTS = ('localhost',
+                   'mediumhost.io',
+                   'malware.badhost.org',
+                   '4-verybadhost.com',
+                   'ads.worsthostever.net')
+
+CLEAN_HOSTS = ('goodhost.gov', 'verygoodhost.com')
+
+URLS_TO_CHECK = ('http://localhost',
+                 'http://mediumhost.io',
+                 'http://malware.badhost.org',
+                 'http://4-verybadhost.com',
+                 'http://ads.worsthostever.net',
+                 'http://goodhost.gov',
+                 'ftp://verygoodhost.com'
+                 'http://qutebrowser.org')
+
+
+@pytest.fixture
+def data_tmpdir(monkeypatch, tmpdir):
+
+    """Set tmpdir as datadir"""
+
+    tmpdir = str(tmpdir)
+    monkeypatch.setattr('qutebrowser.utils.standarddir.data', lambda: tmpdir)
+
+
+# XXX Why does read_hosts needs basedir to be None
+# in order to print message 'run adblock-update to read host blocklist' ?
+# browser/adblock.py line 138
+# TODO Replace with unittest mock.mock / See test webelem
+class BaseDirStub:
+
+    """Mock for objreg.get('args') called in adblock.HostBlocker.read_hosts."""
+
+    def __init__(self):
+        self.basedir = None
+
+
+@pytest.yield_fixture
+def basedir():
+
+    """Register a Fake basedir."""
+
+    args = BaseDirStub()
+    objreg.register('args', args)
+    yield args
+    objreg.delete('args')
+
+
+class FakeDownloadItem(QObject):
+
+    """Mock browser.downloads.DownloadItem."""
+
+    finished = pyqtSignal()
+
+    def __init__(self, fileobj, name):
+        super().__init__()
```

Comment:
Please also add a `parent` argument at the end and pass that to the `QObject` constructor. Even if it's not needed here, it's just good practise.